packages/react/src/jsx-runtime.tsx

```diff
@@ -34,18 +34,21 @@ type JSXFactory = (
    */
   componentFactory: (props: any) => ServerModelState | ReactElement,
   { children, ...passedProps }: Record<string, any>,
-  key?: string
+  key?: string,
+  isStaticChildren?: boolean
 ) => ServerModelState
 
 export const jsxs: JSXFactory = (componentFactory, passedProps, key) => {
-  Object.defineProperty(passedProps.children, '$static', {
-    value: true,
-    enumerable: false,
-  })
-  return jsx(componentFactory, passedProps, key)
+  return jsx(componentFactory, passedProps, key, true)
 }
 
-export const jsx: JSXFactory = (componentFactory, passedProps, key) => {
+export const jsx: JSXFactory = (componentFactory, passedProps, key, isStaticChildren) => {
```

Comment:
This PR is ready for review, but I need to change this to `jsxImpl`, as `jsx` itself as called by Babel or other transformers does not receive this boolean argument, so this is a bit misleading.

We will have jsx, jsxs, jsxDEV and jsxImpl (the base function), just like React.